dali/aux/optical_flow/turing_of/optical_flow_turing_test.cc

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <dali/aux/optical_flow/turing_of/optical_flow_turing.h>
+#include <dali/test/dali_test_config.h>
+#include <dali/util/cuda_utils.h>
+#include <opencv2/opencv.hpp>
+#include <cuda_runtime.h>
+#include <gtest/gtest.h>
+#include <fstream>
+#include <vector>
+#include <string>
+
+namespace dali {
+namespace optical_flow {
+namespace testing {
+
+namespace {
+constexpr float kFlowVectorEpsilon = 1.f / 32;
+}
+
+
+TEST(OpticalFlowTuringTest, DecodeFlowVectorTest) {
+  // (In)sanity test
+  using std::vector;
+  vector<int16_t> test_data = {101, -32376, 676, 3453, -23188};
+  vector<float> ref_data = {3.15625f, -1011.75f, 21.125f, 107.90625f, -724.625f};
+  for (size_t i = 0; i < test_data.size(); i++) {
+    EXPECT_NEAR(ref_data[i], kernel::decode_flow_component(test_data[i]), kFlowVectorEpsilon);
+  }
+}
+
+
+TEST(OpticalFlowTuringTest, BgrToAbgrSynteticTest) {
+  std::vector<uint8_t> data = {
+          73, 5, 47, 71, 30, 1,
+          80, 41, 60, 60, 85, 41,
+          55, 66, 4, 94, 59, 47,
+          64, 83, 96, 61, 30, 95,
+          88, 95, 63, 96, 78, 16,
+          81, 89, 81, 2, 18, 35,
+  };
+  std::vector<uint8_t> reference = {
+          255, 73, 5, 47, 255, 71, 30, 1, 0, 0,
+          255, 80, 41, 60, 255, 60, 85, 41, 0, 0,
+          255, 55, 66, 4, 255, 94, 59, 47, 0, 0,
+          255, 64, 83, 96, 255, 61, 30, 95, 0, 0,
+          255, 88, 95, 63, 255, 96, 78, 16, 0, 0,
+          255, 81, 89, 81, 255, 2, 18, 35, 0, 0,
+  };
+  size_t width = 6, pitch = 10, height = 6;
+
+  uint8_t *input, *tested;
+  CUDA_CALL(cudaMallocManaged(&input, data.size()));
```

Comment:
You don't need unified memory for input if you copy the data using cudaMemcpy anyway.